synth-shell-prompt/synth-shell-prompt.sh

```diff
@@ -195,9 +195,17 @@ prompt_command_hook()
 
 	## CHOOSE PS1 FORMAT IF INSIDE GIT REPO
 	if [ ! -z "$(getGitBranch)" ] && $SSP_GIT_SHOW; then
-		PS1=$SSP_PS1_GIT
+		if [ -n "${VIRTUAL_ENV}" ]; then
+			PS1="$SSP_PS1_GIT \e[0;31m(`basename ${VIRTUAL_ENV}`)\e[0m "
```

Comment:
I think it would be more elegant if you did firts a GIT check. Pseudo code:
```
If in git, then:
  PS1 = PS1_GIT
if not:
  PS1 = PS1_NORMAL
```

And then, afterward, also check if in a virtual env.
```
If in virtual env:
  PS1 = PS1 + virtual_env_stuff
if not:
  leave as is:
```

This reduces code duplication. But it might only work if all you do us _append_ extra stuff in case you are in a virtual env. You probably have a better insight than me into this. So feel free to ignore this suggestions

Reply:
Hmm. Thinking about it. Can you be in a virtual env and a git repository at the same time? Never used virtual envs before. If not, then you could not:
```
if in git...
else if in virtual_env....
else, normal...
```

Reply:
You can be in a virtual env and not in a git repo, basically python will set the env var VIRTUAL_ENV if your currently in one, otherwise it's empty (so it's not exactly like git where it's directory dependant, it's env var dependant).